torch/utils/dlpack.py

```diff
@@ -25,3 +15,38 @@
 The dlpack shares the tensors memory.
 Note that each dlpack can only be consumed once.
 """)
+
+
+def from_dlpack(ext_tensor: Any) -> torch.Tensor:
+    """from_dlpack(ext_tensor) -> Tensor
+
+    Convers a tensor from a external library into a ``torch.Tensor``
+    by means of the ``__dlpack__`` protocol.
+
+    The tensor will share the memory with the object represented
+    in the dlpack.
+
+    In order to keep backward compatibility, this function also admits
+    to pass a dlpack capsule object.
+    Note that each dlpack capsule can only be consumed once.
```

Comment:
What happens if `from_dlpack()` is called on the same dlpack capsule more than once?

Reply:
something like a double free errror is likely to happen. I think the C layer checks this.